applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -400,18 +320,15 @@ static void stream_stopped_cb(struct bt_audio_stream *stream)
 
 	LOG_INF("Stream %p stopped", (void *)stream);
 
-	ret = channel_index_get(NULL, stream, &channel_index);
+	ret = channel_index_get(stream->conn, &channel_index);
 	if (ret) {
-		LOG_ERR("Stream not found");
+		LOG_ERR("Channel index not found");
 	} else {
 		atomic_clear(&headsets[channel_index].iso_tx_pool_alloc);
 	}
 
-	if (headsets[AUDIO_CH_L].sink_stream->ep->status.state != BT_AUDIO_EP_STATE_STREAMING
-#if !CONFIG_STREAM_BIDIRECTIONAL
-	    && headsets[AUDIO_CH_R].sink_stream->ep->status.state != BT_AUDIO_EP_STATE_STREAMING
-#endif
-	) {
+	if (headsets[AUDIO_CH_L].sink_stream.ep->status.state != BT_AUDIO_EP_STATE_STREAMING &&
+	    headsets[AUDIO_CH_R].sink_stream.ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
```

Comment:
Is it OK to remove the bidir flag here?

Reply:
It should be, because now we can have bidir using 2 headsets. I also improved this by creating the "ep_state_check()" function